Remove 'session_state' from tokens (#28354)
keycloak/keycloak: docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc

 * `auth_time`

This helps to reduce even more the number of claims in a lightweight access token, but also gives the chance to configure claims that were always added automatically.

= Removed `session_state` claim

The `session_state` claim which contains the same value as the `sid` claim is now removed from all tokens as it is not required according to the OpenID Connect Front-Channel Logout and OpenID Connect Back-Channel Logout specifications. The `session_state` claim remains present in the Access Token Response in accordance with OpenID Connect Session Management specification.

> A new `Session State (session_state)` mapper is also included and can be assigned to client scopes (for instance `basic` client scope) to revert to the old behavior.

Reviewer: Is it possible to also include a small notice that method `setSessionState()` is removed from `IDToken` class and `getSessionState()` is made deprecated? It is possible that some people might use this?

Reviewer: Thanks! I see there are test failures. I guess they are related to the fact that our `keycloak.js` adapter is relying on the `session_state` claim from the token, which causes javascript adapter to not behave correctly. Looks it is here: https://github.com/keycloak/keycloak/blob/main/js/libs/keycloak-js/src/keycloak.js#L1012 .

Is it possible to fix this in `keycloak.js` and also mention in the upgrading guide another minor note that people using old version of `keycloak.js` adapter in their applications must add the `SessionStateMapper` to their clients? We did similar already for the `nonce` parameter in the file `changes-25_0_0.adoc`, so maybe that section can be updated somehow to handle both `nonce` and `session_state` ?

Reviewer 2: Thanks for sharing that bit of documentation. It looks like it wasn't adopted in the end? According to https://www.keycloak.org/2024/06/keycloak-2500-released and anecdotal experience, session_state isn't included in any claims (including the access token) these days.

 * `auth_time`

> This helps to reduce even more the number of claims in a lightweight access token, but also gives the chance to configure claims that were always added automatically.

Reviewer 3: ```suggestion
This provides additional help to reduce the number of claims in a lightweight access token, but also gives the chance to configure claims that were always added automatically.
```

Author: Thanks, Done!

= Removed `session_state` claim

> The `session_state` claim which contains the same value as the `sid` claim is now removed from all tokens as it is not required according to the OpenID Connect Front-Channel Logout and OpenID Connect Back-Channel Logout specifications. The `session_state` claim remains present in the Access Token Response in accordance with OpenID Connect Session Management specification.

Reviewer 3: ```suggestion
The `session_state` claim, which contains the same value as the `sid` claim, is now removed from all tokens as it is not required according to the OpenID Connect Front-Channel Logout and OpenID Connect Back-Channel Logout specifications. The `session_state` claim remains present in the Access Token Response in accordance with OpenID Connect Session Management specification.
```

Author: Thanks, Done!

> Note that the `setSessionState()` method is also removed from `IDToken` class in favor of the `setSessionId()` method, and the `getSessionState()` method is now deprecated.

Reviewer 3: ```suggestion
Note that the `setSessionState()` method is also removed from the `IDToken` class in favor of the `setSessionId()` method, and the `getSessionState()` method is now deprecated.
```
